Python/compile.c

```diff
@@ -8181,6 +8213,102 @@ add_checks_for_loads_of_uninitialized_variables(basicblock *entryblock,
     return 0;
 }
 
+static int
+expand_del_noerror(basicblock *entryblock, int none_oparg)
+{
+    for (basicblock *b = entryblock; b != NULL; b = b->b_next) {
+        int room_needed = 0;
```

Comment:
This function would be simpler if you use insert_instruction(). Is it for optimization that you aren't?

Reply:
Yeah, I was trying to avoid quadratic time, but I suppose that could only happen if one long basicblock had many `DELETE_FAST_CHECK` instructions, e.g.

```
def f(x):
    del x; del x; del x; del x; del x  
    del x; del x; del x; del x; del x  
    del x; del x; del x; del x; del x  
```

But such code is useless, so I'll try to clean this up.

The `DELETE_NOERROR` instructions should typically happen at basicblock boundaries, so I believe there shouldn't be any performance issues there.